publisher/internal/analytics/ethereum/process.go

```diff
@@ -22,65 +12,34 @@ func (a *Analytics) ProcessMessage(msg analytics.Message, send analytics.Sender)
 	if !hasTopics(eLog) {
 		return nil
 	}
-	// 1) All events are put into cache
-	a.addLogToTxCache(eLog)
 
-	if !a.isMintEvent(eLog) || (a.isMintEvent(eLog) && !isUniswapPositionsNFT(eLog.Data)) {
-		return nil
-	}
+	a.addLogToTxCache(eLog) //All events are put into cache
 
-	// 2) Mint event is found
-	var liquidityEntry LiquidityAddition
-	txEventsFromCache, _ := a.eventLogCache.Get(eLog.TransactionHash)
-	for _, evLog := range txEventsFromCache.([]EventLog) { // Go through all events of this transaction
-		if reflect.DeepEqual(evLog, eLog) { // This allows to correctly process many Mint events in one transaction
-			break
-		}
-		if a.isAvoidEvent(evLog) {
-			liquidityEntry = LiquidityAddition{} // Reset gathered records
-			continue
-		}
-		if a.isTransferEvent(evLog) { // 3) Searching for relevant "Transfer" event(s)
-			liquidityEntry = a.handleLiquidityTransfer(evLog, liquidityEntry) // 4) Decoding / expanding "Transfer" events
-		}
-	}
-	liquidityEntry = a.checkAndUpdateMissingToken(eLog, liquidityEntry)         // 5) Adding missing token if only 1 token transfer was made
-	liquidityEntry = a.includeTokenPrices(liquidityEntry)                       // 6) Getting token prices
-	liquidityEntry, doPublish := a.finalizeLiquidityEntry(eLog, liquidityEntry) // 7) Save Liquidity Entry and Liquidity Pool
-
-	if !doPublish {
+	var operation Operation
+	switch {
+	case a.isBurnEvent(eLog):
+		operation = NewRemovalOperation(a.db, a.eventLogCache, a, send)
+	case a.isMintEvent(eLog):
+		operation = NewAdditionOperation(a.db, a.eventLogCache, a, send)
+	default:
 		return nil
 	}
-	message, err := makeMessage(liquidityEntry)
+
+	var err error
+	operation, err = operation.Extract(eLog)
 	if err != nil {
-		log.Println(err)
-		return err
+		log.Printf("%s\n", err)
```

Comment:
Either use `err.Error()`, or use `%v`. Also, some description about the nature of the error would be helpful.